npm/pkg/controlplane/goalstateprocessor/goalstateprocessor.go

```diff
@@ -324,9 +336,7 @@ func (gsp *GoalStateProcessor) processPolicyRemoveEvent(goalState *protos.GoalSt
 
 func validatePayload(payload map[string]*protos.GoalState) bool {
 	for _, v := range payload {
-		if len(v.GetData()) != 0 {
-			return true
-		}
+		return len(v.GetData()) != 0
```

Comment:
this will only reach the first item in the slice

Reply:
That is intentional, payload is invalid if all items are at 0, else it is valid.